src/som/vmobjects/SObjectWithContext.java

```diff
@@ -0,0 +1,16 @@
+package som.vmobjects;
+
+import com.oracle.truffle.api.frame.MaterializedFrame;
+
+/**
+ * Something that has a context.
+ * Used by SBlock and SClass currently.
+ */
+
+public interface SObjectWithContext {
+
+  MaterializedFrame getContext();
```

Comment:
Please specify the semantics for both methods.

We need to get better with documentation.